test/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/BecomeLeadControllerSpec.scala

```diff
@@ -492,62 +467,41 @@ class BecomeLeadControllerSpec
+            mockRetrieveEmailAddressAndSendEmail(eori1, None, "removedAsLeadToOldLead", undertaking1, undertakingRef, None)(Right(EmailSent))
             mockDelete[UndertakingJourney](eori4)(Right(()))
             mockSendAuditEvent[BusinessEntityPromotedSelf](
               AuditEvent.BusinessEntityPromotedSelf(undertakingRef, "1123", eori1, eori4)
             )
           }
 
           checkPageIsDisplayed(
-            performAction()(lang),
+            performAction()(lang.code),
             messageFromMessageKey("become-admin-confirmation.title")
           )
         }
 
         "when user has selected English language" in {
-          testDisplay(
-            "template_promoted_themself_as_lead_email_to_lead_EN",
-            "template_removed_as_lead_email_to_previous_lead_EN",
-            English.code
-          )
+          testDisplay(English)
         }
 
         "when user has selected Welsh language" in {
-          testDisplay(
-            "template_promoted_themself_as_lead_email_to_lead_CY",
-            "template_removed_as_lead_email_to_previous_lead_CY",
-            Welsh.code
-          )
-
+          testDisplay(Welsh)
         }
 
       }
 
       "redirect to next page" when {
 
         "become lead journey doesn't contains accept terms" in {
-          val newLeadParams =
-            SingleEORIEmailParameter(eori4, undertaking1.name, undertakingRef, "promotedAsLeadToNewLead")
-          val oldLeadParams =
-            SingleEORIEmailParameter(eori1, undertaking1.name, undertakingRef, "removedAsLeadToOldLead")
           inSequence {
             mockAuthWithNecessaryEnrolment(eori4)
             mockGet[BecomeLeadJourney](eori4)(Right(newBecomeLeadJourney.some))
             mockRetrieveUndertaking(eori4)(undertaking1.some.toFuture)
             mockAddMember(undertakingRef, businessEntity4.copy(leadEORI = true))(Right(undertakingRef))
             mockAddMember(undertakingRef, businessEntity1.copy(leadEORI = false))(Right(undertakingRef))
-            mockRetrieveEmail(eori4)(Right(RetrieveEmailResponse(EmailType.VerifiedEmail, validEmailAddress.some)))
-            mockSendEmail(validEmailAddress, newLeadParams, "template_promoted_themself_as_lead_email_to_lead_EN")(
-              Right(EmailSendResult.EmailSent)
-            )
-            mockRetrieveEmail(eori1)(Right(RetrieveEmailResponse(EmailType.VerifiedEmail, validEmailAddress.some)))
-            mockSendEmail(validEmailAddress, oldLeadParams, "template_removed_as_lead_email_to_previous_lead_EN")(
-              Right(EmailSendResult.EmailSent)
-            )
+            mockRetrieveEmailAddressAndSendEmail(eori4, None, "promotedAsLeadToNewLead", undertaking1, undertakingRef, None)(Right(EmailSent))
+            mockRetrieveEmailAddressAndSendEmail(eori1, None, "removedAsLeadToOldLead", undertaking1, undertakingRef, None)(Right(EmailSent))
```

Comment:
I was using them separate to test if the correct language  template get picked when the language is toggled. In the new test case you can see the template name is not being used and no way to test if correct one is getting picked. If that looks fine , then I can approve this PR :)

Reply:
Thanks! There's a little more refactoring to come that will address this.